lab_6_pipeline/pipeline.py

```diff
@@ -17,16 +39,58 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError  # built-in
+
+        files = list(self.path_to_raw_txt_data.iterdir())
+        if not files:
+            raise EmptyDirectoryError
+
+        metas = [file for file in files if 'meta.json' in file.name]
+        raws = [file for file in files if 'raw.txt' in file.name]
+        if len(metas) != len(raws):
+            raise InconsistentDatasetError
+
+        metas.sort(key=lambda x: int(get_article_id_from_filepath(x)))
+        raws.sort(key=lambda x: int(get_article_id_from_filepath(x)))
+
+        for index, (meta, raw) in enumerate(zip(metas, raws)):
+            if not meta.exists() or not raw.exists():
+                raise FileNotFoundError  # built-in
+
+            if index + 1 != get_article_id_from_filepath(meta) \
+                    or index + 1 != get_article_id_from_filepath(raw)\
+                    or meta.stat().st_size == 0 or raw.stat().st_size == 0:
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        files = [file for file in self.path_to_raw_txt_data.iterdir()
```

Comment:
use glob with pattern instead of condition

Reply:
Ok